src/robotFactory.js

```diff
@@ -1,10 +1,83 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {
+    x: 0, y: 0, z: 0,
+  }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    if (typeof this.coords.x === 'undefined') {
+      this.coords.x = 0;
+    }
 
-class DeliveryDrone {}
+    if (typeof this.coords.y === 'undefined') {
+      this.coords.y = 0;
+    }
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return 'Robot: ' + this.name + ', ' + 'Chip version: '
+      + this.chipVersion + ', ' + 'Weight: ' + this.weight;
+  }
```

Comment:
Use template string + concatenation for good solution.
```suggestion
  getInfo() {
    return `Robot: ${this.name}, ` 
    + `Chip version: ${this.chipVersion}, `
    + `Weight: ${this.weight}`
  }
```